pkg/cmd/fixtures/templates/with-resource-complex.toml

```diff
@@ -0,0 +1,48 @@
+[metadata]
+  name = "experiment"
+
+[global]
+  plan = "plan"
+  case = "case"
+  builder = "docker:go"
+  runner = "local:docker"
+
+{{ with (load_resource "./resource-complex.toml" ) }}
+{{- range .groups }}
+[[groups]]
+  id = "{{ .id }}"
+  instances = { count = 1 }
+
+  [groups.build]
+    selectors = ['{{ .selector }}']
+
+  [groups.build_config]
+    build_base_image = 'golang:{{ .go_version }}-buster'
```

Comment:
This looks very unwieldy and prone to typos. I would prefer the much cleaner option of using something like `go_build_image=golang:1.18-buster` in the `resource.toml` file itself.

Reply:
So the discussion is whether the benefit of reusing a single variable (go_version) in multiple locations is worth the complexity it adds to the template definition. That might be fine in some cases, and not fine in other cases.

But that's something a Testground User might worry about.

We're wearing a Testground Maintainer hat here: we're not using or maintaining this composition file, we're only testing the templating feature with a few fixtures that looks like real tests. I believe the test would be as good or worst if we dropped this part because it would cover a different, smaller, use case.

Reply:
Understood. Feel free to close this.